File: examples/utility-r/create-example-visuals.R
#' This script provides a demo on how to generate example visuals using the 'vivainsights' R library

# Set up -------------------------------------------------------------------------------------------
# Load packages
library(tidyverse)
library(vivainsights)

# Set output visual path 
out_vis <- here::here("examples", "utility-r", "example-visuals")

# SVG output: `create_rank()` -----------------------------------------------------------------------

pq_data %>%
    create_rank(
        metric = "Collaboration_hours",
        hrvar = c("Organization", "LevelDesignation", "FunctionType", "SupervisorIndicator"),
        return = "plot") %>%
    export(
        method = "svg",
        path = paste(out_vis, "create_rank.svg", sep = "/"),
        timestamp = FALSE
        )

# SVG output: `create_trend()` ----------------------------------------------------------------------

pq_data %>%
    create_trend(
        metric = "Email_hours",
        hrvar = "LevelDesignation"
    ) %>%
    export(
        method = "svg",
        path = paste(out_vis, "create_trend", sep = "/"),
        timestamp = FALSE
        )

# SVG output: `create_bar()` ------------------------------------------------------------------------

pq_data %>%
    create_bar(
        metric = "Internal_network_size",
        hrvar = "Organization"
    ) %>%
    export(
        method = "svg",
        path = paste(out_vis, "create_bar", sep = "/"),
        timestamp = FALSE
        )      

# SVG output: `create_boxplot()` --------------------------------------------------------------------

pq_data %>%
    create_boxplot(
        metric = "Collaboration_hours",
        hrvar = "Organization"
    ) %>%
    export(
        method = "svg",
        path = paste(out_vis, "create_boxplot", sep = "/"),
        timestamp = FALSE
        )

# SVG output: `create_scatter()` --------------------------------------------------------------------

pq_data %>%
    create_scatter(
        metric_x = "Email_hours",
        metric_y = "Meeting_hours",
        hrvar = "FunctionType"
    ) %>%
    export(
        method = "svg",
        path = paste(out_vis, "create_scatter", sep = "/"),
        timestamp = FALSE
        )

# SVG output: `create_bubble` --------------------------------------------------------------------        

pq_data %>%
    create_bubble(
        metric_x = "Email_hours",
        metric_y = "Meeting_hours",
        hrvar = "FunctionType"
    ) %>%
    export(
        method = "svg",
        path = paste(out_vis, "create_bubble", sep = "/"),
        timestamp = FALSE
        )

# SVG output: `keymetrics_scan()` -------------------------------------------------------------------

pq_data %>%
    keymetrics_scan() %>%
    export(
        method = "svg",
        path = paste(out_vis, "keymetrics_scan", sep = "/"),
        timestamp = FALSE
        )

# SVG output: `network_g2g()` -----------------------------------------------------------------------

g2g_data %>%
    network_g2g(
        metric = "Meeting_Count",
        primary = "PrimaryCollaborator_Organization",
        secondary = "SecondaryCollaborator_Organization",
        exc_threshold = 0.05
        ) %>%
    export(
        method = "svg",
        path = paste(out_vis, "network_g2g", sep = "/"),
        timestamp = FALSE
        )        

# SVG output: `network_p2p()` -----------------------------------------------------------------------

set.seed(100)
p2p_data <- p2p_data_sim()

p2p_data %>%
    network_p2p(
        style = "ggraph",
        return = "plot"
    ) %>%
    export(
        method = "svg",
        path = paste(out_vis, "network_p2p", sep = "/"),
        timestamp = FALSE
        )      

File: examples/utility-python/create-example-visuals.py
"""
Viva Insights - Python Utility Script
================================
This script provides a demo on how to generate example visuals using the 'vivainsights' Python library
All outputs are saved as SVG in a local output folder that you can customize the path to. 
"""

# Setup ---------------------------------------------------------------------------------------------
# load packages
import vivainsights as vi
import os
import matplotlib.pyplot as plt

# load in-built datasets
pq_data = vi.load_pq_data() # load and assign in-built person query
g2g_data = vi.load_g2g_data() # load and assign in-built group-to-group query
p2p_data = vi.p2p_data_sim(size = 400) # Simulate a person-to-person query, using the `size` parameter to control for size

# set output path
out_path = os.getcwd() + '/examples/utility-python/example-visuals/'

# SVG output: `create_bar()` ------------------------------------------------------------------------

plot_bar = vi.create_bar(data=pq_data, metric='Emails_sent', hrvar='Organization', mingroup=5)

vi.export(
    plot_bar,
    file_format = 'svg',
    path = out_path + 'create_bar',
    timestamp = False
)

# SVG output: `create_line()` -----------------------------------------------------------------------

plot_line = vi.create_line(data=pq_data, metric='Emails_sent', hrvar='Organization', mingroup=5, return_type='plot')

vi.export(
    plot_line,
    file_format = 'svg',
    path = out_path + 'create_line',
    timestamp = False
)

# SVG output: `create_boxplot()` --------------------------------------------------------------------

plot_box = vi.create_boxplot(data=pq_data, metric='Emails_sent', hrvar='Organization', mingroup=5, return_type='plot')

vi.export(
    plot_box,
    file_format = 'svg',
    path = out_path + 'create_boxplot',
    timestamp = False
)

# SVG output: `create_rank()` -----------------------------------------------------------------------

plot_rank = vi.create_rank(
    data=pq_data,
    metric='Collaboration_hours',
    hrvar = ['Organization', 'FunctionType', 'LevelDesignation', 'SupervisorIndicator'],
    mingroup=5,
    return_type = 'plot'
)

vi.export(
    plot_rank,
    file_format = 'svg',
    path = out_path + 'create_rank',
    timestamp = False
)

# SVG output: `network_g2g()` -----------------------------------------------------------------------

plot_g2g = vi.network_g2g(
    data = g2g_data,
    primary = "PrimaryCollaborator_Organization",
    secondary = "SecondaryCollaborator_Organization",
    return_type = "plot"
    )

# Workaround whilst # 15 is being fixed
# vi.export(
#     plot_g2g,
#     file_format = 'svg',
#     path = out_path + 'network_g2g',
#     timestamp = False
# )

# SVG output: `network_p2p()` -----------------------------------------------------------------------

plot_p2p = vi.network_p2p(data = p2p_data, return_type = "plot") 

# Workaround whilst # 15 is being fixed
# vi.export(
#     plot_p2p,
#     file_format = 'svg',
#     path = out_path + 'network_p2p',
#     timestamp = False
# )